lors01-sp

farmaextra1 opened the pull request:
Většina souborů z projektu. Vše by mělo být funkční. Design musím dodělat.

Odkaz: https://kombuchaeshop.infinityfreeapp.com/

farmaextra1 committed:
Kombucha-eshop

Většina souborů z projektu. Vše by mělo být funkční. Ještě zítra otestuji a přídám odkaz na web.

farmaextra1 committed:
lors01-sp

Update

nvbach91 commented on www/lors01/resources/js/Pages/CheckoutPage.jsx:
```diff
@@ -0,0 +1,187 @@
+import '../../css/checkoutPage.css';
+
+export default function CheckoutPage() {
+    const defaultFormData = {
```
predvyplnit formular s udaji prihlaseneho uzivatele

nvbach91 reviewed (changes requested):
- uzivatel po dokonceni nakupu neni presmerovan na hlavni stranku a zustava na strance checkout a tim padem se dalo vytvorit prazdne objednavky
- nezobrazovat obsah objednavky v seznamu objednavek, ale v samostatne strance nebo modalovem okne
- zobrazit zpusob dopravy a ceny v seznamu objednavek
- jak se dostanu do administrace e-shopu kvuli napr. sprave produktu a kategorii?
- chybi navigace do hlavni stranky
- existuje strankovani? muzete pridat 50 dalsich produktu?
- co se stane se skladem kdyz dva uzivatele nakoupi ve stejnem okamziku vetsi pocet nez je skladem
- zadne emaily nedostavam, ani do spamu

farmaextra1 commented:
Dobrý den, upravím. Dal jsem vám admin práva, můžete jít nyní do administrátorské stránky

farmaextra1 commented:
Jestli stránkováním myslíte, že když bude více produktů na hlavní stránce tak se zbytek schová pod další stránky tak to není, jelikož jsem k tomu přistupoval spíše z realistického hlediska, kdy tento eshop není wholesale ecommerce. Přidat dalších 50 produktů bych mohl přes productseeder.

farmaextra1 commented:
chybi navigace do hlavni stranky. Kde myslíte? U pokladny a platby jsem odendal header, jelikož je to poté spíše distrakce, která zákazníka může rozptýlit od dokončení objednávky.

farmaextra1 committed:
lors01-sp

Changes

farmaextra1 commented:
Platba kartou je na stripe test mode, takze se muze pouzit 4242 4242 4242 4242 08/25 111, akorat nyni pres hosting nefunguje webhook, pravdepodobne z duvodu pouziti free ssl, ktere nema intermediate ca. Na ngroku funguje normalne. Emaily se mi posílají také, takže to jsem neměnil, ale zajímá mě, kde je chyba. Redirect jsem přidal i do controlleru, předtím byl pouze na straně klienta, ale funguje mi v obou případech. Nyní není možné udělat další prázdnou objednávku, disablenul jsem buttony když jsou cartItems prázdné a přidal jsem header do order confirmation stranky. Pridal jsem take kontrolu skladu pred objednavkou, aby se zabranilo vytvoreni objednavky, kdyby ji provedli dva uzivatele za sebou, odstrani se jim z localstorage a vezme je to na homepage. Je tam take modal pro zobrazeni detailu objednavky, jinak je ukazana pouze cena, status a zpusob dopravy. Formular s udaji prihlaseneho uzivatele se prevyplni nyni take.

farmaextra1 committed:
lors01-sp

Obhajoba + sql

nvbach91 commented on www/lors01/app/Http/Controllers/OrderController.php:
```diff
@@ -12,8 +12,8 @@ class OrderController extends Controller
     {
-        $userEmail = $request->user()->email;
-        $orders = Order::where('email', $userEmail)
+        $userId = $request->user()->user_id;
```
tohle nakonec nema byt `->id`?

farmaextra1 committed:
Update

Dobrý den, přidal jsem serverovou validaci checkout inputů. Z projektu bych potřeboval 26 bodů - o 1 bod více než nyní.
